spring-integration-file/src/main/java/org/springframework/integration/file/FileReadingMessageSource.java

```diff
@@ -495,12 +499,14 @@ public boolean isRunning() {
 							files.remove(file);
 						}
 						else {
-							if (file.isDirectory()) {
-								files.addAll(walkDirectory(file.toPath()));
-							}
-							else {
-								files.remove(file);
-								files.add(file);
+							if (file.exists()) {
+								if (file.isDirectory()) {
+									files.addAll(walkDirectory(file.toPath()));
+								}
+								else {
+									files.remove(file);
+									files.add(file);
+								}
 							}
```

Comment:
We should probably add a debug log here "File removed since watch event received".